pydantic/fields.py

```diff
@@ -60,6 +60,7 @@ class _FromFieldInfoInputs(typing_extensions.TypedDict, total=False):
     decimal_places: int | None
     union_mode: Literal['smart', 'left_to_right'] | None
     discriminator: str | types.Discriminator | None
+    deprecated: str | None
```

Comment:
I think you've added support for this below logic wise, but we probably want to change the type hint to something similar to `str | warnings.deprecated | typing_extensions.deprecated | None` as @dmontagu originally suggested...